google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/BaseEntity.java

```diff
@@ -446,18 +447,21 @@ public B setNull(String name) {
   BaseEntity(Builder<K, ?> builder) {
     this.key = builder.key;
     this.properties = ImmutableSortedMap.copyOf(builder.properties);
+    this.serializedSize = this.toPb().getSerializedSize();
```

Comment:
what's the cost of calculating this for a large entity? Most of the time we don't need this value. Perhaps this could be done in demand and not stored or only lazily stored in the field?

Reply:
The cost could be varied it depends on user entered properties. Does this look fine? 
```java
/** Returns the number of bytes in the serialized protobuf. */
  public int getSerializedSize() {
    if (serializedSize == 0) {
      serializedSize = this.toPb().getSerializedSize();
    }
    return serializedSize;
  }
```
Here we are calculating entity size when demand.